src/base/SolutionArray.cpp

```diff
@@ -553,83 +1219,132 @@ void SolutionArray::readEntry(const std::string& fname, const std::string& id)
     }
 
     // restore remaining data
-    for (const auto& name : names) {
-        if (!states.count(name)) {
-            vector_fp data = file.readVector(id, name, m_size);
-            m_extra.emplace(name, data);
+    if (m_meta.hasKey("components")) {
+        const auto& components = m_meta["components"].asVector<string>();
+        bool back = false;
+        for (const auto& name : components) {
+            if (hasComponent(name) || name == "X" || name == "Y") {
+                back = true;
+            } else {
+                addExtra(name, back);
+                AnyValue data;
+                data = file.readData(path, name, m_dataSize);
+                setComponent(name, data);
+            }
+        }
+        m_meta.erase("components");
+    } else {
+        // data format used by Python h5py export (Cantera 2.5)
```

Comment:
Adding a `user_warning`. I think that we should support the legacy format indefinitely - users may have stored HDF data in legacy format; not being able to read it after Cantera 3.0 would be problematic.

As an aside, renaming the transport model from `Mix` to `mixture-averaged` raises a separate deprecation warning. Some of the deprecated default names should probably be kept around (e.g. by using a `addLegacyAlias` rather than a `addDeprecatedAlias`).

Reply:
That's a reasonable distinction -- user code is something that needs to be continuously updated to keep pace with new Cantera versions, but it would be a different thing to say that you need to update your data files (without a specific migration path as in the case of CTI/XML to YAML).

I wasn't necessarily planning on removing the deprecated model names anytime soon, and the warning doesn't specify that these will be removed after a particular Cantera version, unlike our warnings for deprecated functions/classes.